tests/Schema/TableTest.php

```diff
@@ -372,24 +377,50 @@ public function testAddForeignKeyConstraintUnknownLocalColumnThrowsException():
         $table->addForeignKeyConstraint($foreignTable->getName(), ['foo'], ['id']);
     }
 
-    public function testAddForeignKeyConstraint(): void
+    /** @throws Exception */
+    public function testAddForeignKeyConstraintWithInvalidMatchType(): void
     {
         $table = new Table('foo');
-        $table->addColumn('id', Types::INTEGER);
+        $table->addColumn('bar_id', Types::INTEGER);
 
-        $foreignTable = new Table('bar');
-        $foreignTable->addColumn('id', Types::INTEGER);
+        $this->expectException(ValueError::class);
```

Comment:
I didn't bother to introduce a dedicated exception for these errors because eventually `addForeignKeyConstraint()` should be replaced with using `TableEditor` and `ForeignKeyConstraintEditor`, and the latter accepts the match type and referential actions as enums.